runtime/src/accounts_index.rs

```diff
@@ -204,30 +204,94 @@ impl<T: 'static + Clone + IsCached> WriteAccountMapEntry<T> {
         })
     }
 
+    fn addref(item: &AtomicU64) {
+        item.fetch_add(1, Ordering::Relaxed);
+    }
+
+    pub fn upsert_new_key<'a>(
```

Comment:
Because this function performs exactly the definition of  "upsert" , maybe rename to just `upsert()`